subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/HierarchicalFileWatcherUpdater.java

```diff
@@ -146,13 +141,13 @@ private void updateWatchedHierarchies(SnapshotHierarchy root) {
             return;
         }
         if (!hierarchiesToStopWatching.isEmpty()) {
-            fileWatcher.stopWatching(hierarchiesToStopWatching.stream()
+            getFileWatcher().stopWatching(hierarchiesToStopWatching.stream()
                 .map(Path::toFile)
                 .collect(Collectors.toList())
             );
         }
         if (!hierarchiesToStartWatching.isEmpty()) {
-            fileWatcher.startWatching(hierarchiesToStartWatching.stream()
+            getFileWatcher().startWatching(hierarchiesToStartWatching.stream()
```

Comment:
Why not keep these as `protected` fields instead?

Reply:
Good point. Done.